flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -200,4 +241,204 @@ private static DataType handleLogicalTypeSchema(Schema fieldSchema) {
+                                    + precision
+                                    + ", it only supports precision less than equal to 6.");
+                }
+
+                // use int to represents Datetime, we only support millisecond/microsecond when
+                // deserialization
+                Schema datetime = avroLogicalDateTimeType.addToSchema(avroDateTimeType);
+                return nullable ? nullableSchema(datetime) : datetime;
+            case DECIMAL:
+                DecimalType decimalType = (DecimalType) logicalType;
+                precision = decimalType.getPrecision();
+                // store BigDecimal as byte[]
+                Schema decimal =
+                        LogicalTypes.decimal(precision, decimalType.getScale())
+                                .addToSchema(SchemaBuilder.builder().bytesType());
+                if (precision <= 38 && precision > 0) {
+                    decimal.addProp("isNumeric", true);
+                }
+                return nullable ? nullableSchema(decimal) : decimal;
+            case ROW:
+                RowType rowType = (RowType) logicalType;
+                List<String> fieldNames = rowType.getFieldNames();
+                // we have to make sure the record name is different in a Schema
+                SchemaBuilder.FieldAssembler<Schema> builder =
+                        SchemaBuilder.builder().record(rowName).fields();
+                for (int i = 0; i < rowType.getFieldCount(); i++) {
+                    String fieldName = fieldNames.get(i);
+                    org.apache.flink.table.types.logical.LogicalType fieldType =
+                            rowType.getTypeAt(i);
+                    SchemaBuilder.GenericDefault<Schema> fieldBuilder =
+                            builder.name(fieldName)
+                                    .type(convertToSchema(fieldType, rowName + "_" + fieldName));
+                    builder = fieldBuilder.noDefault();
+                }
+                Schema record = builder.endRecord();
+                return nullable ? nullableSchema(record) : record;
+            case MULTISET:
+            case MAP:
+                Schema map =
+                        SchemaBuilder.builder()
+                                .map()
+                                .values(
+                                        convertToSchema(
+                                                extractValueTypeToAvroMap(logicalType), rowName));
+                return nullable ? nullableSchema(map) : map;
+            case ARRAY:
+                ArrayType arrayType = (ArrayType) logicalType;
+                Schema array =
+                        SchemaBuilder.builder()
+                                .array()
+                                .items(convertToSchema(arrayType.getElementType(), rowName));
+                return nullable ? nullableSchema(array) : array;
+            case RAW:
+
```

Comment:
Unhandled case. Either add some handling logic or remove this case altogether

Reply:
Removed.